packages/mwp-core/src/renderers/server-render.jsx

```diff
@@ -277,26 +277,29 @@ const makeRenderer = (
 		};
 		const checkReady = state =>
 			state.preRenderChecklist.every(isReady => isReady);
-		const populateStore = store =>
-			new Promise((resolve, reject) => {
-				// dispatch SERVER_RENDER to kick off API middleware
-				store.dispatch({ type: SERVER_RENDER, payload: url });
+		const populateStore = store => {
+			// keep this above the SERVER_RENDER so
+			// state.flags is available to query calls
+			if (checkReady(store.getState())) {
```

Comment:
`checkReady` is only used _after_ `SERVER_RENDER` is dispatched - the store can't be ready before any API calls are made.